parcels/particleset.py

```diff
@@ -295,7 +298,12 @@ def ctypes_struct(self):
         class CParticles(Structure):
             _fields_ = [(v.name, POINTER(np.ctypeslib.as_ctypes_type(v.dtype))) for v in self.ptype.variables]
 
-        cdata = [np.ctypeslib.as_ctypes(self.particle_data[v.name]) for v in self.ptype.variables]
+        def cdata_for(v):
+            cdata_array = np.ctypeslib.as_ctypes(self.particle_data[v.name])
+            # cast to bare pointer
+            return cast(cdata_array, POINTER(np.ctypeslib.as_ctypes_type(v.dtype)))
```

Comment:
so, question for my understanding: are now all particle cdata pointers-of-base types ?

Reply:
That's correct

Reply:
k - sad that it makes thing that bit more complex to keep in mind, but I agree with your reasoning: this way seems the most applicable solution.